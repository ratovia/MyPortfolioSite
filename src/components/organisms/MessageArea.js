import React from "react"
import styled from "styled-components"
import { TypographyH5, TypographyH4 } from "../atoms/Typography"
import { SNSIcons } from "../molecules/SNSIcons"
import { Photograph } from "../atoms/Photograph"
const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: flex-start;
  flex-basis: 100px;
  padding: 100px;
`

const PhotoArea = styled.div`
  width: 465px;
  margim-right: 100px;
`

const TextArea = styled.div`
  width: 570px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin-left: 30px;
`

const MessageItem = styled.div`
  margin-bottom: 60px;
`

export const MessageArea = ({ imageUrl, visionText }) => {
  return (
    <Container>
      <PhotoArea>
        <Photograph imageUrl={imageUrl} radius></Photograph>
      </PhotoArea>
      <TextArea>
        <MessageItem>
          <TypographyH4
            label="昨日の自分より今日の自分を意識して日々学んでいます。"
            color="white"
          ></TypographyH4>
        </MessageItem>
        <MessageItem>
          <TypographyH5
            label="この先やってみたいこと"
            color="white"
          ></TypographyH5>
          <TypographyH5 label={visionText} color="white"></TypographyH5>
        </MessageItem>
        <MessageItem>
          <SNSIcons></SNSIcons>
        </MessageItem>
      </TextArea>
    </Container>
  )
}
