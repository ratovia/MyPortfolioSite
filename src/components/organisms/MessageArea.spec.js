import React from "react"
import renderer from "react-test-renderer"
import { MessageArea } from "./MessageArea"
import javascriptIcon from "../../stories/assets/javascript_icon.png"

describe("MessageArea", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MessageArea visionText="Hello world" imageUrl={javascriptIcon} />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
