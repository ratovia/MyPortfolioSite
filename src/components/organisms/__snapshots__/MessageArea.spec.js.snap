// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageArea renders correctly 1`] = `
<div
  className="sc-iCfLBT ehlSFw"
>
  <div
    className="sc-furvIG cBgrlr"
  >
    <img
      alt="アイキャッチ"
      className="sc-gKckTs dDzAHu"
      height="100%"
      src="test-file-stub"
      width="100%"
    />
  </div>
  <div
    className="sc-pVTma eUwiAj"
  >
    <div
      className="sc-jrQzUz OEohN"
    >
      <h4
        className="sc-gsDJrp gKGDLI"
      >
        <div>
          <p
            className="sc-bdvvaa kxpcMG"
            color="white"
          >
            昨日の自分より今日の自分を意識して日々学んでいます。
          </p>
        </div>
      </h4>
    </div>
    <div
      className="sc-jrQzUz OEohN"
    >
      <h5
        className="sc-dkPtyc liUvOS"
      >
        <p
          className="sc-bdvvaa kxpcMG"
          color="white"
        >
          この先やってみたいこと
        </p>
      </h5>
      <h5
        className="sc-dkPtyc liUvOS"
      >
        <p
          className="sc-bdvvaa kxpcMG"
          color="white"
        >
          Hello world
        </p>
      </h5>
    </div>
    <div
      className="sc-jrQzUz OEohN"
    >
      <div
        className="sc-jRQAMF bSpCWo"
      >
        <img
          alt="アイコン"
          className="sc-eCImvq flMHMz"
          height="48px"
          src="test-file-stub"
          width="48px"
        />
        <img
          alt="アイコン"
          className="sc-eCImvq flMHMz"
          height="48px"
          src="test-file-stub"
          width="48px"
        />
        <img
          alt="アイコン"
          className="sc-eCImvq flMHMz"
          height="48px"
          src="test-file-stub"
          width="48px"
        />
        <img
          alt="アイコン"
          className="sc-eCImvq flMHMz"
          height="48px"
          src="test-file-stub"
          width="48px"
        />
      </div>
    </div>
  </div>
</div>
`;
